server/src/main/java/org/opensearch/plugins/SecureTransportSettingsProvider.java

```diff
@@ -36,6 +37,33 @@ default Collection<TransportAdapterProvider<Transport>> getTransportAdapterProvi
         return Collections.emptyList();
     }
 
+    /**
+     * Returns parameters that can be dynamically provided by a plugin providing a {@link SecureTransportSettingsProvider}
+     * implementation
+     * @param settings settings
+     * @return an instance of {@link SecureTransportParameters}
+     */
+    default Optional<SecureTransportParameters> parameters(Settings settings) {
+        return Optional.of(new DefaultSecureTransportParameters(settings));
+    }
+
+    interface SecureTransportParameters {
+        boolean dualModeEnabled();
+    }
+
+    class DefaultSecureTransportParameters implements SecureTransportParameters {
```

Comment:
Could you please move it off as package protected class (since by default class are public in the interface)
```suggestion
    class DefaultSecureTransportParameters implements SecureTransportParameters {
```

Reply:
Should the access modifier be public here or move this to a different file altogether?

Reply:
>  move this to a different file altogether?

Right, we cannot change visibility in the interface, so we have to move it off to a different file

Reply:
Moved to a separate file in the same package